### PR TITLE
Update Ansible to 5.2 and Python to 3.10

### PR DESCRIPTION
**Pull request by asciible**

### CODE REVIEW

File: cli/licenses.py [hunk 1]
```diff
@@ -1681,16 +1639,5 @@
...
-        "License text": "MIT License\n\nCopyright (c) [year] [fullname]\n\nPermission is hereby granted, free of charge, to any person obtaining a copy\nof this software and associated documentation files (the \"Software\"), to deal\nin the Software without restriction, including without limitation the rights\nto use, copy, modify, merge, publish, distribute, sublicense, and/or sell\ncopies of the Software, and to permit persons to whom the Software is\nfurnished to do so, subject to the following conditions:\n\nThe above copyright notice and this permission notice shall be included in all\ncopies or substantial portions of the Software.\n\nTHE SOFTWARE IS PROVIDED \"AS IS\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR\nIMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,\nFITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE\nAUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER\nLIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,\nOUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE\nSOFTWARE.\n"
     }
 ]
```
Comment (erzetpe) [inline cli/licenses.py]: Please, add enter at the end of file.

File: .devcontainer/pyproject.toml [hunk 2]
```diff
@@ -5,15 +5,15 @@ description = "Epiphany Dev Container"
...
+python = "^3.10"
 pyyaml = "*"
 jinja2 = "*"
 boto3 = "*"
 jsonschema = "*"
 python-json-logger = "*"
-ansible = "==2.10.7"
 "ruamel.yaml" = "*"
 azure-cli = "^2.32.0"
+ansible = "^5.2.0"
```
Comment (erzetpe) [inline .devcontainer/pyproject.toml]: Maybe this may be be more specific, like using tilde instead of caret?

**Review by erzetpe — CHANGES_REQUESTED** [review]

File: cli/licenses.py (hunk 1 above)
Comment (seriva) [reply cli/licenses.py]: This is automatically generated each time we update it, so lets leave as is.

File: .devcontainer/pyproject.toml (hunk 2 above)
Comment (seriva) [reply .devcontainer/pyproject.toml]: Agreed, use '''poetry add ansible@5.2.0''' to fix it in place please. We only want to update this periodically, not every time we update the dependencies.

Comment (asciible) [reply .devcontainer/pyproject.toml]: Do we want to lock python to 3.10 as well? It was specified as ^3.7

Comment (seriva) [reply .devcontainer/pyproject.toml]: For me its ok, as this is coming from the container anyway which is fixed. What I did see is that I also did not lock the azure-cli version when I did the terraform update. Could you pls also remove and add azure-cli in again in this pull locking it to 2.32.0?

`poetry add azure-cli@2.32.0`

File: cli/licenses.py (hunk 1 above)
Comment (erzetpe) [reply cli/licenses.py]: We need to fix this and replace the line:  licenses_content = textwrap.dedent(licenses_content) + json.dumps(all_deps_data, indent=4) in file: .devcontainer/gen-dependency-info.py with for example:  licenses_content = f"{textwrap.dedent(licenses_content)}{json.dumps(all_deps_data, indent=4)}\n". This enter at the end of file is there for a reason.

File: .devcontainer/pyproject.toml (hunk 2 above)
Comment (erzetpe) [reply .devcontainer/pyproject.toml]: This can be a good idea to lock this to python 3.10 as this may be additional safety net.

Comment (seriva) [reply .devcontainer/pyproject.toml]: So lets lock it.

**Comment by atsikham** [comment]
Changelog

**Review by erzetpe — DISMISSED** [review]

File: docs/changelogs/CHANGELOG-2.0.md [hunk 3]
```diff
@@ -17,6 +17,8 @@
   - Azurerm provider 1.38.0 to 2.91.0
   - AWS provider 2.26 to 3.71.0
   - Upgraded Azure-cli 2.29 to 2.32
+  - Ansible-core 2.10 to 2.12 (Ansible 5.2.0)
```
Comment (seriva) [inline docs/changelogs/CHANGELOG-2.0.md]: Just name it Ansible whater to 5.2.0 and the entries under the root issue #2847

File: cli/licenses.py (hunk 1 above)
Comment (seriva) [reply cli/licenses.py]: @asciible pls consider adding the extra new line fix suggested by @erzetpe

**Review by erzetpe — APPROVED** [review]

**Review by seriva — APPROVED** [review]

**Pull request merged**